frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -650,6 +661,147 @@ describe("AddToDashSelectDashModal", () => {
+
+          // There's no way to math a URL that a query param is not present
+          // with fetch-mock, so we have to assert it manually.
+          const call = fetchMock.lastCall("path:/api/search");
+          const urlObject = new URL(checkNotNull(call?.request?.url));
+          expect(urlObject.pathname).toEqual("/api/search");
+          expect(
+            urlObject.searchParams.get("filter_items_in_personal_collection"),
+          ).toEqual(null);
+        });
+      });
+
+      describe("questions in public collections", () => {
+        it("should search dashboards only in public collections", async () => {
+          await setup({
+            card: CARD_IN_PUBLIC_COLLECTION,
+          });
+
+          const typedText = "question";
+          const dashboardInPublicCollection = createMockSearchResult({
+            name: "dashboard in public collection",
+            model: "dashboard",
+          });
+          fetchMock.get(
+            {
+              url: "path:/api/search",
+              query: {
+                models: "dashboard",
+                q: typedText,
+              },
+            },
+            {
+              data: [dashboardInPublicCollection],
+            },
+          );
+
+          userEvent.click(screen.getByRole("button", { name: "Search" }));
+          userEvent.type(
+            screen.getByPlaceholderText("Search"),
+            "question{enter}",
+          );
+
+          expect(
+            await screen.findByText(dashboardInPublicCollection.name),
+          ).toBeInTheDocument();
+
+          // There's no way to math a URL that a query param is not present
+          // with fetch-mock, so we have to assert it manually.
+          const call = fetchMock.lastCall("path:/api/search");
+          const urlObject = new URL(checkNotNull(call?.request?.url));
+          expect(urlObject.pathname).toEqual("/api/search");
+          expect(
+            urlObject.searchParams.get("filter_items_in_personal_collection"),
+          ).toEqual(null);
+        });
+      });
+
+      describe("questions in personal collections", () => {
+        it("should search all dashboards", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+          });
+
+          const typedText = "question";
+          const dashboardInPublicCollection = createMockSearchResult({
+            id: 1,
+            name: "dashboard in public collection",
+            model: "dashboard",
+          });
+          const dashboardInPersonalCollection = createMockSearchResult({
+            id: 2,
+            name: "dashboard in personal collection",
+            model: "dashboard",
+          });
+          fetchMock.get(
+            {
+              url: "path:/api/search",
+              query: {
+                models: "dashboard",
+                q: typedText,
+                filter_items_in_personal_collection: "only",
+              },
+            },
```

Comment:
Good suggestion. As a benefit, both checking for the absence of `filter_items_in_personal_collection` and the presence of it would have the same structure, which is a plus.

Reply:
Fixed in 1129fa738fa3d223a99efb3e115132f96b76f085